core/types/access_witness.go

```diff
@@ -67,6 +67,7 @@ func (aw *AccessWitness) SetLeafValue(addr []byte, value []byte) {
 	var stem [31]byte
 	copy(stem[:], addr[:31])
 
+	fmt.Printf("SetLeafValue %x\n", addr)
 	if chunk, exists := aw.Chunks[common.BytesToHash(addr)]; exists {
 		chunk.value = value
```

Comment:
there is a problem here, because Go sucks: the `[]` call will return a copy of the chunk, not a reference. So you end up not modifying the target value.

I am replacing this with:
```suggestion
		chunk.value = value
		aw.Chunks[common.BytesToHash(addr)] = chunk
```

and that fixes the problem